### PR TITLE
Corrupted branch...

### LINKED ISSUE #56: precipitation widget html
Create precipitation rain cloud with animated rain drops using html, css, and js.

### PR DESCRIPTION
**Pull request by abijeet-dhillon**
## In this PR:

- Created a precipitation widget which is an animation of a rain cloud. In the future, I want to adjust the amount of rain falling depending on the amount of precipitation fallen, but that is going to be done after we decide how we want to measure our stats (daily, every 6 hours, etc.).

- Changed the colours of the search bar to fit the dark mode theme better on both the fire and weather pages.

- Page toggle button sticks to the bottom right of the pages.

Closes #56 #146 #129

### CODE REVIEW

File: Site/bc_weather_station_dashboard/website/static/CSS/precipitation_card.css [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    width: 150px;
+    height: 40px;
+    top: 30px;
+    background: #fff;
+    border-radius: 100px;
+    filter: drop-shadow(8px 8px 0 #0005);
+    animation: animateCloud 3s linear infinite;
+}
+
+@keyframes animateCloud {
```
Comment (robertyac) [inline Site/bc_weather_station_dashboard/website/static/CSS/precipitation_card.css:26]: Maybe only flash the cloud when it is really down pouring?

**Review by robertyac — DISMISSED** [review]
Wow that's crazy love this. 

one thing I've been noticing is that sometimes the value is sticky for rain? maybe take a look at map.js but I think what is happening is that when there is no new value it just takes the previous value instead of resetting it to 0.

https://github.com/carsondrobe/Fellas/assets/112982183/250ff13b-67c0-43f2-a812-00c92b3c7d3b

**Review by carsondrobe — DISMISSED** [review]
Looks absolutely wonderful. I am also getting the sticky bug Robert mentioned where if you try to click around to different weather stations, the precipitation text doesn't update.

**Comment by carsondrobe** [comment]
> Looks absolutely wonderful. I am also getting the sticky bug Robert mentioned where if you try to click around to different weather stations, the precipitation text doesn't update.


https://github.com/carsondrobe/Fellas/assets/91719000/a3120b46-70d7-4112-9ea6-01da352760d3

**Comment by abijeet-dhillon** [comment]
> Looks absolutely wonderful. I am also getting the sticky bug Robert mentioned where if you try to click around to different weather stations, the precipitation text doesn't update.

Fixed it, resets to 0 mm upon new clicks then updates the widget with a new value if available (since the database doesn't have a N/A value, it only says 0 for precipitation).

File: Site/bc_weather_station_dashboard/website/static/CSS/precipitation_card.css (hunk 1 above)
Comment (abijeet-dhillon) [reply Site/bc_weather_station_dashboard/website/static/CSS/precipitation_card.css:26]: Good idea, I'll add that into the issues for this week!

**Comment by abijeet-dhillon** [comment]
> Wow that's crazy love this.
> 
> one thing I've been noticing is that sometimes the value is sticky for rain? maybe take a look at map.js but I think what is happening is that when there is no new value it just takes the previous value instead of resetting it to 0.
> 
>  Screen.Recording.2024-03-21.at.8.48.21.PM.mov

Fixed it, resets to 0 mm upon new clicks then updates the widget with a new value if available (since the database doesn't have a N/A value, it only says 0 for precipitation).

**Pull request closed without merging**